vulkan-cts: 1.3.10.0 -> 1.4.1.3

jljusten opened the pull request:
Vulkan SC (Safety Critical) CTS is separate from the Vulkan CTS. This change splits it out into a separate vulkansc-cts package, and uses the latest Vulkan SC CTS tag, vulkansc-cts-1.0.2.1 with this package.

Then, the Vulkan CTS is upgraded: 1.3.10.0 -> 1.4.1.2.

These changes were prompted by the Vulkan SC executable (deqp-vksc) failing to build with the newer Vulkan CTS version. This build failure is not a bug of the CTS as the vulkan-cts-* tags are for the Vulkan CTS and not the Vulkan SC CTS. In other words, vulkan-cts-* tags support the Vulkan CTS and vulcansc-cts-* tags support the Vulkan SC CTS.

I don't have access to a Vulkan SC setup, so I was only able to test this command (to dump the CTS test names) with the new vulkansc-cts package:
 * `deqp-vksc --deqp-runmode=txt-caselist`

Cc: @Flakebi

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

jljusten committed:
vulkan-cts: 1.3.10.0 -> 1.4.1.3

Note that the Vulkan SC executable deqp-vksc had to be dropped
(previous patch) from the Vulkan CTS (vulkan-cts) package, or this
update would lead to a build failure.

Signed-off-by: Jordan Justen <jljusten@gmail.com>

jljusten committed:
vulkan-cts: Drop deqp-vksc from vulkan-cts package

The Vulkan SC (Safety Critical) CTS has different release tags and
build dependencies from the Vulkan CTS.

When updating the vulkan-cts package version, I found that the newer
version would fail to build if we attempted to build the deqp-vksc
executable.

I suspect that building the Vulkan SC CTS executable, deqp-vksc, may
not be supported unless using a vulkansc-cts-* tag.

This change adds cmake build define SELECTED_BUILD_TARGETS=deqp-vk,
which was introduced in vulkan-cts-1.3.7.0, and reduces the amount of
code built. Notably, it causes the build to no longer attempt to build
deqp-vksc.

Ref: vulkan-cts b22a8f06d ("Allow selecting a subset of targets at configuration time")
Signed-off-by: Jordan Justen <jljusten@gmail.com>

Flakebi commented:
Sounds mostly good to me, thanks!
As the packages are rather similar, it would be good to share the code for them. The `cts-sources.py` script is the same and the `default.nix` is almost identical, so it would be a shame to have the code duplicated :)

Flakebi commented:
It seems like `deqp-vksc` does not work correctly at the moment, so I’d also be fine with just dropping it.
```bash
$ deqp-vksc -n dEQP-VKSC.info.device
…
FATAL ERROR: Couldn't match tags from sub.qpa Subprocess failed with exit code 0(0) at vktTestPackage.cpp:1009
```

jljusten commented:
> It seems like `deqp-vksc` does not work correctly at the moment, so I’d also be fine with just dropping it.
> 
> ```shell
> $ deqp-vksc -n dEQP-VKSC.info.device
> …
> FATAL ERROR: Couldn't match tags from sub.qpa Subprocess failed with exit code 0(0) at vktTestPackage.cpp:1009
> ```

Do you have a driver that supports Vulkan SC? I also see a similar error message, but I don't think my driver supports Vulkan SC.

I did as you suggested, and just dropped Vulkan SC support, rather than creating a new package for it. Perhaps we can add the separate package in the future if we can get a way to confirm that it is working.

I also updated the package from 1.4.1.2 to a new 1.4.1.3 tag.

Flakebi commented:
Seems like we need to remove one of the two leading `\n` here to make nixfmt happy:
https://github.com/NixOS/nixpkgs/blob/28f541ba59a96b9127ff8d7e5100a643e5aca3b4/pkgs/by-name/vu/vulkan-cts/vk-cts-sources.py#L77

Otherwise the change looks good to me (builds and passes tests `nix-build . -A vulkan-cts.tests`), thanks!

jljusten committed:
vulkan-cts: Update vk-cts-sources.py for nixfmt

Signed-off-by: Jordan Justen <jljusten@gmail.com>

Flakebi reviewed (approved):
Thanks, LGTM!

jljusten commented:
@Flakebi: Did I need to do anything further on this PR, or should it automatically get picked up now that it's been approved?

I thought it would merge without me taking further action, but then I saw this:
https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#how-to-merge-pull-requests-yourself

Flakebi commented:
Not much to do from your side, I’ll mention it on Discourse as ready to merge (the discourse bot should cross-reference that here soon).

Flakebi commented:
I opened #399178 to update to 1.4.2.0 (and increase the chance to get one of these merged, as vulkan-cts is currently broken in nixos-unstable).
